src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -630,6 +630,47 @@ bool is_object_superset(
       pj{newer})));
 }
 
+bool is_enum_superset(json::Value const& older, json::Value const& newer) {
+    auto older_it = older.FindMember("enum");
+    auto newer_it = newer.FindMember("enum");
+    auto older_is_enum = older_it != older.MemberEnd();
+    auto newer_is_enum = newer_it != newer.MemberEnd();
+
+    if (!older_is_enum && !newer_is_enum) {
+        // both are not an "enum" schema, compatible
+        return true;
+    }
+
+    if (!(older_is_enum && newer_is_enum)) {
+        // only one is an "enum" schema, not compatible
+        return false;
+    }
+
+    // both "enum"
+    // check that all "enum" values of newer are present in older.
+    auto older_set = older_it->value.GetArray();
+    auto newer_set = newer_it->value.GetArray();
+
+    if (newer_set.Size() > older_set.Size()) {
+        // quick check:
+        // newer has some value not in older
+        return false;
+    }
+
+    // TODO: current implementation is O(n^2), but could be O(n) with normalized
+    // input json
+    for (auto& v : newer_set) {
+        // NOTE: values equality is performed with an O(n^2) search, this can
+        // also be improved with normalization of the input
+        if (older_set.end() == std::ranges::find(older_set, v)) {
+            // newer has an element not in older
+            return false;
+        }
+    }
+
+    return true;
```

Comment:
nitpick: I find it slightly easier to read like this: 
```suggestion
    return std::ranges::all_of(newer_set, [&older_set](const auto& v) {
        // NOTE: values equality is performed with an O(n^2) search, this can
        // also be improved with normalization of the input
        auto contains = older_set.end() != std::ranges::find(older_set, v);
        return contains;
    });
```

Reply:
hopefully this will be just a temp solution